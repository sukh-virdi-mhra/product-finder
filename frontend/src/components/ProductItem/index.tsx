import React, { useState, FormEvent } from "react"
import IProduct from "../../models/product"
import styled from "styled-components"

const StyledProductItem = styled.div`
  font-family: Calibri;
  color: red;
  font-size: 20px;
  font-weight: 900;
  border-style: solid;
  margin: 20px;
  padding: 10px;
`

interface IProductItem {
  item: IProduct
}

const ProductItem: React.FC<IProductItem> = (props) => {
  function handleClick() {
    console.log("Button clicked")
      const requestOptions = {
        method: 'POST',
        headers: {
            'Content-Type': 'application/json',
            'Authorization': 'Bearer my-token',
            'My-Custom-Header': 'foobar'
        },
        body: JSON.stringify({ title: 'React POST Request Example' })
    };
    fetch("/order", { method: "POST" })
  }

  return (
    <StyledProductItem>
      <p>{props.item.plNumber}</p>
      <p>{props.item.name}</p>
      <p>{props.item.dose}</p>
      <p>{props.item.substance}</p>
      <p>{props.item.url}</p>
      <button type="button" onClick={handleClick}>
        Order
      </button>
    </StyledProductItem>
  )
}

export default ProductItem