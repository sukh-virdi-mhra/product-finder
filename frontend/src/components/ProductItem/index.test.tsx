import React from "react"
import { shallow, mount } from "enzyme"
import ProductItem from "./index"

let Product = {
    plNumber: "PL001730050",
    name: "Ibugel gel",
    dose: "10mg",
    substance: "Ibuprofen",
    url: "https://mhraproductsproduction.blob.core.windows.net/docs/54454db7a69e9153ab7302232cdb8ee9230579ca"
}

describe("ProductItem", () => {
  it("renders shallow", () => {
    const component = shallow(<ProductItem item={Product} />)
    expect(component).toMatchSnapshot()
  }),
    it("renders mount", () => {
      const component = mount(<ProductItem item={Product} />)
      expect(component).toMatchSnapshot()
    })
})