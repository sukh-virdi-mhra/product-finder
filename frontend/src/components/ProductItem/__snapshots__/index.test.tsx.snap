// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductItem renders mount 1`] = `
<ProductItem
  item={
    Object {
      "dose": "10mg",
      "name": "Ibugel gel",
      "plNumber": "PL001730050",
      "substance": "Ibuprofen",
      "url": "https://mhraproductsproduction.blob.core.windows.net/docs/54454db7a69e9153ab7302232cdb8ee9230579ca",
    }
  }
>
  <styled.div>
    <div
      className="sc-AxjAm klYsxZ"
    >
      <p>
        PL001730050
      </p>
      <p>
        Ibugel gel
      </p>
      <p>
        10mg
      </p>
      <p>
        Ibuprofen
      </p>
      <p>
        https://mhraproductsproduction.blob.core.windows.net/docs/54454db7a69e9153ab7302232cdb8ee9230579ca
      </p>
      <button
        onClick={[Function]}
        type="button"
      >
        Order
      </button>
    </div>
  </styled.div>
</ProductItem>
`;

exports[`ProductItem renders shallow 1`] = `
<styled.div>
  <p>
    PL001730050
  </p>
  <p>
    Ibugel gel
  </p>
  <p>
    10mg
  </p>
  <p>
    Ibuprofen
  </p>
  <p>
    https://mhraproductsproduction.blob.core.windows.net/docs/54454db7a69e9153ab7302232cdb8ee9230579ca
  </p>
  <button
    onClick={[Function]}
    type="button"
  >
    Order
  </button>
</styled.div>
`;
